src/mds/MDSAuthCaps.h

```diff
@@ -300,17 +299,23 @@ class MDSAuthCaps
     }
   }
 
-  bool root_squash_in_caps() const {
-    for (const MDSCapGrant &g : grants) {
-      if (g.match.root_squash) {
-        return true;
+  bool root_squash_in_caps(std::string_view fs_name) const {
+    for (const MDSCapGrant& g : grants) {
+      if (fs_name_match(g.match.fs_name, fs_name)) {
+        if (g.match.root_squash) {
+          return true;
+        }
       }
     }
     return false;
   }
 
   friend std::ostream &operator<<(std::ostream &out, const MDSAuthCaps &cap);
   std::string to_string();
+protected:
+  static bool fs_name_match(std::string_view capfs, std::string_view other) {
```

Comment:
IMO it will be better to add this method to `MDSCapMatch`? We can then call `g.match.match_fsname(fsname)`.

Reply:
Sure, done.